feat(dashboard): add Arabic translation

zaaakher opened the pull request:
This PR adds Arabic localization. 

Ideally, we might need to add `dir="rtl"` conditionally somewhere to flip the UI to make it RTL

zaaakher committed:
chore: add arabic translation

MEClouds commented on packages/admin/dashboard/src/i18n/languages.ts:
```diff
@@ -1,7 +1,13 @@
+  {
+    code: "ar",
+    display_name: "عربي",
+    ltr: true,
```
change LTR to false (because Arabic use RTL)

MEClouds commented on packages/admin/dashboard/src/i18n/languages.ts:
```diff
@@ -1,7 +1,13 @@
+    code: "ar",
+    display_name: "عربي",
+    ltr: true,
+    date_locale: arSA,
```
use ar for generic Arabic,
date_locale: ar ,

MEClouds commented on packages/admin/dashboard/src/i18n/languages.ts:
```diff
@@ -1,7 +1,13 @@
 export const languages: Language[] = [
+  {
+    code: "ar",
+    display_name: "عربي",
```
display_name: "العربية",

zaaakher committed:
feat(dashboard): set `ltr` to false + add changeset

zaaakher commented:
@MEClouds Done 👍

shahednasser committed:
docs-util: fix links in references to current page (#10703)

shahednasser committed:
docs-util: fix query params not retrieved correctly for some routes (#10708)

shahednasser committed:
docs: split commerce modules sidebars (#10705)

shahednasser committed:
fix(medusa,types,js-sdk): fix request query parameter types for store product routes (#10707)

* fix(medusa,types): fix request query parameter types for store product routes

* fix test errors

shahednasser committed:
docs: fix typo in extend core flow (#10709)

* docs: fix typo in extend core flow

* small fixes

shahednasser committed:
docs: split events reference across modules (#10711)

* docs: split events reference across modules

* add lint exceptions

shahednasser committed:
docs: split admin widget zone to pages (#10719)

ranjithkumar8352 committed:
fix(medusa-cli): fix medusa develop command crashing on file change on windows (#10717)

* fix(medusa-cli): fix medusa develop command crashing on file change on windows

* chore: update changeset to patch

shahednasser committed:
docs: improve link docs for commerce modules (#10726)

shahednasser committed:
docs: fix multiple typos in customization chapters (#10727)

shahednasser committed:
docs: add workflow / step tags (#10731)

shahednasser committed:
fix(core-flows): export updateTaxRegionsStep (#10732)

Export `updateTaxRegionsStep` to ensure it's included in generated references

sradevski committed:
feat(medusa): Add health endpoint for all running modes of medusa (#10737)

shahednasser committed:
docs: revise commerce modules overview pages (#10738)

* revise API Key Module overview

* revise auth module

* support ref sidebar items

* remove examples

* revise cart module

* revise currency

* revise customer module

* revise fulfillment module

* revise inventory module

* revise order module

* revise payment

* revise pricing module

* revise product module

* revise promotion module

* revise region module

* revise sales channel module

* revise stock location module

* revise store module

* revise tax module

* revise user module

* lint content + fix snippets

shahednasser committed:
chore(js-sdk): add tags to js-sdk (#10739)

shahednasser committed:
docs: add JS SDK to commerce modules sidebars (#10740)

shahednasser committed:
chore: improve payment provider tsdocs + generate reference (#10742)

shahednasser committed:
docs: fix and improve details on sending authenticated requests with API token (#10744)

* docs: fix and improve details on sending authenticated requests with API token

* fix security schema

shahednasser committed:
docs-util: add publishable api key header parameter to store routes OAS (#10747)

* docs-util: add publishable api key header parameter to store routes OAS

* fix oas

shahednasser committed:
docs: document how to calculate shipping prices in storefront (#10748)

shahednasser committed:
docs: allow scroll with keys (#10752)

shahednasser committed:
docs: update and fix UI props docs (#10754)

shahednasser committed:
feat(create-medusa-app): allow passing project name on command line (#10755)

Allow passing the project name on the command line:

```bash
npx create-medusa-app@latest my-project
```

thetutlage committed:
feat: deprecate remote link (#10768)

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

ranjithkumar8352 committed:
docs: use correct env variable for medusa admin in server mode deployment (#10764)

ahrbil committed:
docs: update desktop nav menu item title (#10125)

Update menu item title to reflect current state of the desktop sidebar

shahednasser committed:
docs: document missing events in reference (#10779)

ranjithkumar8352 committed:
docs: update fulfillment provider module docs to reflect CalculatedShippingOptionPrice (#10778)

The return value below the example for the `calculatePrice` method is shown correctly as `CalculatedShippingOptionPrice` but the example is not updated in the guide "How to Create a Fulfillment Provider Module"

This PR updates the example in the guide and the TS doc

shahednasser committed:
docs: fix code tabs in production (#10784)

shahednasser committed:
docs: fix selected variant in storefront guide (#10785)

shahednasser committed:
chore: fix links to contribution guidelines (#10792)

riqwan committed:
fix(pricing): pricing context calculations only takes into account existing rule attributes (#10771)

* fix(pricing): pricing context calculations only takes into account existing rule attributes

* chore: add changeset

shahednasser committed:
docs: fix highlights in checkout storefront guide (#10793)

shahednasser committed:
docs: fix cross-project link for Card components (#10794)

thetutlage committed:
feat: deprecate remoteQueryConfig in favor of queryConfig (#10773)

Fixes: FRMW-2783

thetutlage committed:
feat: generate modules mappings at runtime (#10791)

thetutlage committed:
Fix/product variants filter (#10808)

Fixes: FRMW-2854

shahednasser committed:
docs: handle product deletion in digital products recipe (#10811)

alsherif-khalaf commented:
What is missing here ?

shahednasser committed:
feat(create-medusa-app): improve by adding depth to clone commands (#10812)

Improve command by passing `--depth 1` to decrease download size + improve for slow speed.

Closes TRI-742

adrien2p committed:
fix(): Workflow cancellation + gracefully handle non serializable state (#10674)

FIXES FRMW-2852

**What**
A workflow distributed transaction expect any response and error to be serializable. When it is not the case, the distributed transaction might fail during the save checkpoint that occurs for async steps. This can lead to unexpected behaviour.

With this pr, we introduce a way to handle non serialazable object in a more sustainable manner, this means the following:

- If a workflow throw any non serialazable error (e.g AWS error that contains full IncomingMessage object that related to network communication, think of req/res) then we identify that this object is not serialzable and we clean up the object to make it serializable without loosing the main information, add a new error to the workflow to informed of this issue and can be handled by the user.
- If a response is not serializable (which should not happen at this point because it is handled before by the value resolver), in that case, we wont be able to reuse that response to continue the workflow which means that the workflow is in a non runnable state. In that case we throw a specific error stating that a non serializable context is being provided

**second what**
This pr refactor the `runAsStep` to add better support for workflow cancelation, especially async ones

daykrm committed:
fix(core-flow): invalid update quantity in update line item in cart workflow (#10405)

* fix: invalid update quantity in update line item in cart workflow

* test: update cart workflow test

* fix: rm shallow copy in transform

MedusaNick committed:
Update README.md (#10825)

 - Renewed the description of What is Medusa
 - Removed Roadmap link as this seem out-of-date

riqwan committed:
fix(promotion): don't evaluate rule condition if conditions to evaluate is empty (#10795)

shahednasser committed:
docs: fix storefront sections in digital product recipe (#10833)

shahednasser committed:
chore: small tsdoc changes for remoteLink and remoteQueryConfig (#10837)

shahednasser committed:
chore(ui): general updates and fixes to TSDocs and types (#10753)

shahednasser committed:
docs: update next.js storefront snippets (#10840)

sradevski committed:
feat(auth-google,auth-github): Allow passing a custom callbackUrl to … (#10829)

* feat(auth-google,auth-github): Allow passing a custom callbackUrl to oauth providers

* feat: Add state management in auth providers

* chore: Changes based on PR review

riqwan committed:
fix(stock-location,core-flows,types): updates existing address when updating stock location (#10832)

* fix(stock-location,core-flows,types): updates existing address when updating stock location address

* chore: use hasOne instead of hasMany

riqwan committed:
feat(dashboard,core-flows,types,utils,medusa): Order cancelations will refund payments (#10667)

* feat(order, types): Add Credit Line to order module

* chore: add action to inject credit lines

* WIP

* chore: add fixes + observe

* chore: fix balances

* chore: add canceled badge

* chore: fix i18n schema

* chore: remove redunddant query

* chore: add changeset

* chore: add credit lines for all cancel cases

* chore: add accounting total

* chore: address review & cleanup

shahednasser committed:
docs: update storefront development snippets for next.js 15 compatibility (#10843)

DivineStudio committed:
Update page.mdx (#10847)

Updated the if-else example condition for when input.is_active is false.

fPolic committed:
feat(core-flows,fulfillment, fulfillment-manual, types): make fulfillment typings more specific (#10677)

**What**
- attempt to add more specific type definitions around fulfillment provider APIs

---

CLOSES CMRC-814
CLOSES CMRC-816
CLOSES CMRC-817

olivermrbl committed:
chore: Add missing changeset

shahednasser committed:
chore: fix version number in tsdocs (#10856)

shahednasser committed:
fix(create-medusa-app): add default storefront url to auth CORS (#10853)

* fix(create-medusa-app): add default storefront url to auth CORS

* add docs cors to store / admin

shahednasser committed:
docs: document registering user with existing email (#10859)

thetutlage committed:
chore: rename remote-query-entry-points.d.ts to query-entry-points.d.ts (#10831)

fPolic committed:
fix(dashboard): cancel order notifications (#10858)

* fix: cancel order notifications

* fix: label

olivermrbl committed:
chore: Version packages

olivermrbl committed:
chore: Release

shahednasser committed:
docs: rename remoteLink and remoteQueryConfig (#10836)

* docs: rename remoteLink and remoteQueryConfig

* change version number

github-actions[bot] committed:
chore(docs): Update version in documentation (automated) (#10864)

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>

shahednasser committed:
docs: fix divider in API reference + clean up layout (#10861)

* docs: fix divider in API reference + clean up layout

* fix build error

shahednasser committed:
docs: document create-medusa-app new project name argument (#10834)

github-actions[bot] committed:
chore(docs): Generated References (automated) (#10866)

Generated the following references:
- `auth`
- `auth_provider`
- `core_flows`
- `fulfillment`
- `fulfillment_provider`
- `js_sdk`
- `medusa`
- `modules`
- `modules_sdk`
- `order`
- `payment`
- `stock_location_next`
- `stock_location_next_models`
- `types`
- `utils`

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

adrien2p committed:
chore(medusa): Add handler path to the http tracing to be able to group by (#10835)

RESOLVES FRMW-2835

In this PR, we trace HTTP requests using the route pattern and not the request URL. This allows us to aggregate all HTTP requests under a single route.

In terms of implementation, we have to self find the route for a given request, since there is no API in express to do the same and we begin tracing even before the request is handed over to Express.

Since, the route matching happens via RegExp matches, we ensure that this does not add much performance overhead. The matching takes between 0.8ms-1.5ms for various different routes

Co-authored-by: Harminder Virk <1706381+thetutlage@users.noreply.github.com>

shahednasser committed:
docs-util: fixes to OAS generator + generate OAS (#10867)

shahednasser committed:
docs: document dependencies for auth provider (#10865)

* docs: document dependencies for auth provider

* fixes

* document overriding callback url

shahednasser committed:
docs: update recipes following 2.2 release (#10868)

ehtisham-afzal committed:
docs: added Cloudflare R2 guide for Object Storage (#10497)

Co-authored-by: Shahed Nasser <27354907+shahednasser@users.noreply.github.com>

gharbi-mohamed-dev committed:
docs: fix typo (#10870)

shahednasser committed:
docs: added llms.txt (#10873)

shahednasser committed:
docs: general fixes and additions (#10876)

shahednasser committed:
docs-util: fix parsing when function if it has three parameters (#10878)

* docs-util: fix parsing when step when it has three parameters

* fix condition

shahednasser committed:
docs-util: add warning on request body change (#10881)

thetutlage committed:
feat: add default retry strategy for redis (#10880)

Fixes: FRMW-2861

thetutlage committed:
feat: remove dead code and refactor the logic of resolving plugins (#10874)

shahednasser committed:
docs: add inventory kit conceptual guide (#10891)

kasperkristensen committed:
fix(pricing): PriceLists of type Sale should not override lower prices (#10882)

Resolves CMRC-840

shahednasser committed:
docs: fix colon in api key header (#10893)

gharbi-mohamed-dev committed:
docs: fix typo (#10894)

pmc12thsuki committed:
fix: event-bus-redis processor execute event before subscriber are loaded (#10823)

* fix(worker): run worker after application start

* test(event-bus-redis): worker should initiate with autorun:false

---------

Co-authored-by: Suki Wang <suki.wang@shoplineapp.com>
Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

riqwan committed:
fix(pricing): add null conditions for deleted at during price calculations (#10896)

shahednasser committed:
docs: add workflow and js sdk reference pages for commerce modules (#10898)

* docs: add workflow and js sdk reference pages for commerce modules

* resolve build errors

adrien2p committed:
fix(modules): Fix miss leading provider resolution error (#10900)

FIXES SUP-560

**What**
Currently, no matter the error when looking for a provider to exists in the container we are throwing a normalized error stating that the provider does not exists in the container. The issue is that the first initialization of the provider occurs the first time we resolve it, and the error can be that the provider failed to be instanciated for any reason. 

In this pr, we are explicitly checking for the error to be an awilix resolution error to throw the classic error and otherwise we provide the issue why the provider failed to be resolved.

shahednasser committed:
fix(types): add missing inventory_items to input of createProductsWorkflow (#10892)

The `createProductsWorkflow` allows passing an `inventory_items` property to assign the inventory items of a product variant, and we even pass it in the `/admin/products` API route when creating a variant, but the workflow's input type doesn't include that property. This PR adds it to the workflow's input type.

shahednasser committed:
docs-util: fix hook input type showing as __type (#10905)

thetutlage committed:
feat: Merge plugin modules (#10895)

Fixes: FRMW-2858

This PR merge the modules exported by the plugins with the modules defined within the user config. As a result, all modules get loaded without changing the internals of the loader.

However, you cannot disable the module of a plugin by re-adding it to the `modules` array. That is something we should handle separately. 

We've added the breaking change label because of the following fix:
We did broke the ability to completely disable modules in the past pr's, in this pr we re add the ability to disable a module and that this modules does not get loaded at all. ([here](https://github.com/medusajs/medusa/pull/10895/commits/6dd164f783d632c927e0a48c50090dcf7387ccdd))

Co-authored-by: Adrien de Peretti <25098370+adrien2p@users.noreply.github.com>

thetutlage committed:
chore: move build utilities to Compiler class (#10904)

Fixes: FRMW-2866

shahednasser committed:
docs: DX improvements to a workflow / step reference page (#10906)

carlos-r-l-rodrigues committed:
fix(orchestration): avoid retry when finished (#10913)

shahednasser committed:
docs: typo fixes in execute workflow section (#10911)

shahednasser committed:
docs-util: add examples generator for workflows and steps (#10914)

* initial changes

* docs-util: generate examples for workflows and steps

wasims1 commented:
hi could you please merge this pull it's only adding a translation @sgirones  @mrnbpt @riqwan 

thanks!

TheMikeyRoss commented:
Any updates on this one? so that it can be included in the next release

kasperkristensen committed:
feat(inventory,dashboard,types,core-flows,js-sdk,medusa): Improve inventory UX (#10630)

* feat(dashboard): Add UI for bulk editing inventory stock (#10556)

* progress

* cleanup types

* add changeset

* fix 0 values

* format schema

* add delete event and allow copy/pasting enabled for some fields

* add response types

* add tests

* work on fixing setValue behaviour

* cleanup toggle logic

* add loading state

* format schema

* add support for bidirectional actions in DataGrid and update Checkbox and RadioGroup

* update lock

* lint

* fix 404

* address feedback

* update cursor on bidirectional select

ranjithkumar8352 committed:
docs: update breaking changes for google auth storefront (#10924)

Medusa v2.2.0 has a breaking change where the `state` must also be passed to the auth callback.

This is not mentioned in the required actions for breaking changes in the release notes. I spent more than an hour trying to understand why the upgrade was breaking Google auth login even though I followed the breaking changes release notes and the docs.

Please update it in the release notes as well if possible - https://github.com/medusajs/medusa/releases/tag/v2.2.0

Co-authored-by: Shahed Nasser <27354907+shahednasser@users.noreply.github.com>

shahednasser committed:
chore(core-flows,types): add tsdocs for create cart workflow (#10928)

shahednasser committed:
docs: fix validation in restaurant-delivery marketplace recipe (#10925)

kasperkristensen committed:
feat(admin-bundler,admin-vite-plugin,medusa): Add support for loading Admin Extensions from plugins (#10869)

Should not be merged before https://github.com/medusajs/medusa/pull/10895

**What**
- Introduces a new `plugin` command to `admin-bundler`, currently not used anywhere but will be called from `medusa build:plugin`
- Discovers plugins with extensions and add passes the to `admin-vite-plugin`.
- Updates `admin-vite-plugin` so its able to read built admin extensions.

Resolves CMRC-830, CMRC-839

shahednasser committed:
docs: fixes to workflow reference tooling (#10931)

rahul-rocket committed:
[Fix] Update Repository Directory Paths for All Packages (#10910)

This PR updates the `repository.directory` field in the `package.json` files for packages located within the `modules` folder.

thetutlage committed:
Feat/plugin develop (#10926)

Fixes: FRMW-2865

In this PR we add support for developing a plugin in watch mode. During the file change, we re-compile the source code (incrementally), publishes the package, and updates the installations of the plugin. 

We are using `yalc` under the hood and it must be installed as a dev dependency in the plugin project and the main Medusa app.

shahednasser committed:
docs: add a note about promotion types support in Medusa Admin (#10933)

shahednasser committed:
docs: fix authentication callback descriptions to include all query parameters (#10932)

ranjithkumar8352 committed:
docs: remove redundant body parameter in fetch customer (#10934)

thetutlage committed:
feat: add plugin build command (#10935)

Fixes: FRMW-2863

Adds the `plugin:build` command that is used to compile the source code of a plugin for publishing it to a package registry. The command is similar to the `build` command, except it does not copy the `package.json` and the `lock` files to the build output

adrien2p committed:
feat: Scaffold plugin in create-medusa-app (#10908)

RESOLVES FRMW-2862

**What**
This PR enable the `create-medusa-app` CLI to accept a new `--plugin` option to scaffold a plugin. This is complementary to all the plugin commands being created/adjusted separately to that pr.
Also, this pr brings a little refactoring around resource scaffolding, the idea was to contain the refactoring to a little area and not expend it to the entire create-medusa-app package to not disrupt and expand the scope for which the purpose was to introduce the plugin scaffolding capabilities

**Addition**
- medusa project will get their package.json name changed to the project name
- Remove build step from medusa project creation

**Plugin flow**

- in the plugin
  - `npx create-medsa-app --plugin`
  - `yarn dev`
- in the project
  - `yalc add plugin-name`
  - `yarn dev`

Any changes on the plugin will publish, push in the local registry which will fire the hot reload of the app and include the new changes from the plugin

adrien2p committed:
feat(cli): Add plugin:publish and plugin:add commands support (#10938)

RESOLVES FRMW-2864
RESOLVES FRMW-2868

**What**
Add support for the following cli commands:
- `plugin:publish`
- `plugin:add`

shahednasser committed:
chore(core-flows,types): improve tsdocs of workflows [1] (#10940)

riqwan committed:
fix(core-flows): return refunded when all captured payments have been refunded (#10923)

what:

- returns refunded when all captured payments have been refunded

riqwan committed:
fix(dashboard, core-flows): improvements to order page on canceled orders (#10888)

what:

- Remove pending payment for canceled orders
- Hide unfulfilled items for canceled orders
- Disable non refundable payments from being refunded
- Populate refund created_by
- Disable order edit when canceled
- Fix bug https://github.com/medusajs/medusa/issues/10852

RESOLVES CMRC-842

shahednasser committed:
chore(docs): generate references following latest workflows reference changes (#10943)

shahednasser committed:
docs-util: remove duplicate union children (#10942)

shahednasser committed:
docs: add additional data description in generated references (#10944)

kasperkristensen committed:
feat(medusa,framework,cli,admin-bundler): Integrate admin extensions into medusa build:plugin (#10941)

**What**
Calls the `plugin` script from `@medusajs/admin-bundler` as part of `medusa plugin:build`.

shahednasser committed:
docs-util: reorder workflow execution examples (#10955)

* docs-util: reorder workflow execution examples

* reorder again

* add examples header

shahednasser committed:
docs: small typo fix (#10956)

shahednasser committed:
docs: handle step tooltip bug in safari (#10954)

RoxDevvv commented:
i need that

shahednasser committed:
docs-util: add exception for workflow examples modification + show summary instead of comment (#10963)

* docs-util: add exception for workflow examples modification + show summary instead of comment

* add currency and country code in example

* add common IDs

olivermrbl committed:
chore: Use local CLI version in CI (#10982)

* chore: Use local CLI version in CI

* Update test-cli-with-database.yml

carlos-r-l-rodrigues committed:
fix(core-flows): missing variable of when condition (#10958)

shahednasser committed:
chore(core-flows,types): improve TSDocs of product workflows (#10966)

shahednasser committed:
chore(core-flows,types): improve TSDocs of order workflows (#10983)

shahednasser committed:
docs: fixes to database transaction docs (#10978)

adrien2p committed:
feat(core, medusa, cli): Enable migration scripts (#10960)

* feat(core, medusa): Enable migration scripts

* spacing

* rm unnecessary import

* Allow to skip script migration

* fix missing options

* add options

* add tests and small changes

* update

* add checks

* add lock mechanism to be extra safe

* Create six-bears-vanish.md

* update queries

* fix tests

---------

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

sradevski committed:
fix: Include is default billing and shipping fields in the address response (#10970)

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

kasperkristensen committed:
chore(dashboard,icons): Update icons and switch icon in CategoryTree (#10961)

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

carlos-r-l-rodrigues committed:
feat(core-flows): validate hook in cart workflows (#10967)

* feat(core-flows): validate hook

* rm only

adrien2p committed:
fix(index): Add type casting to raw order by (#10899)

**What**
Fix index engine query builder to include column type casting on order by

Co-authored-by: Carlos R. L. Rodrigues <37986729+carlos-r-l-rodrigues@users.noreply.github.com>

thetutlage committed:
feat: add support for loading admin extensions from the source (#10975)

sradevski committed:
fix: Put routes loader error inline (#10912)

sradevski committed:
feat: Improvements to payment module and Stripe provider (#10980)

* fix: Correctly parse Stripe error, remove unused method

* fix: Isolate the payment provider error check function

* fix: Allow passing few extra parameters to Stripe

---------

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

shahednasser committed:
chore(core-flows,types): improve TSDocs of cart workflows and steps (#10962)

* chore(core-flows,types): improve TSDocs of cart workflows and steps

* fix build errors

* fix build error

* fix errors

sradevski committed:
feat: Remove fields from payment models that were leftovers from v1 (#10987)

shahednasser committed:
chore(core-flows,types): improve tsdocs of fulfillment (#10995)

shahednasser committed:
chore(core-flows): improve TSDocs of customer and customer groups workflows / steps (#10993)

* improve customer tsdocs

* finished improving tsdocs

* fix build error

adrien2p committed:
fix(framework): Exclude .d.ts files from the glob search and fix insert query (#10990)

**What**
- Exclude definition file from the glob search string
- Properly generate the insert query string

thetutlage committed:
chore: lazy import admin-bundler and run scripts during publish (#10996)

sradevski committed:
feat: Add support for listing saved payment methods in module and Stripe (#10994)

shahednasser committed:
chore(core-flows): improve TSDocs for payment workflows / steps (#10998)

shahednasser committed:
chore(core-flows,types): improve tsdocs of pricing-related workflows (#11000)

* chore(core-flows,types): improve tsdocs of pricing-related workflows

* fix build error

shahednasser committed:
chore(js-sdk): update TSDocs of auth callback (#11005)

* chore(js-sdk): update TSDocs of auth callback

* generate doc

riqwan committed:
feat(promotion,dashboard,types,utils,medusa): Add statuses to promotions (#10950)

what:

- adds a status column to promotion table
- introduce active promotion query
- scope revert, register and compute actions to active promotions
- admin to create and update promotion with statuses

RESOLVES CMRC-845
RESOLVES CMRC-846
RESOLVES CMRC-847
RESOLVES CMRC-848
RESOLVES CMRC-849
RESOLVES CMRC-850

fPolic committed:
fix(dashboard): delete user messages (#11004)

CLOSES CMRC-760

shahednasser committed:
chore: fix manage customer groups of customer OAS (#11011)

* chore: fix manage customer groups of customer OAS

* fix validation error

shahednasser committed:
chore(core-flows,types): improve TSDocs of inventory-related workflows (#11013)

shahednasser committed:
fix(core-flows,medusa): use deleteRefundReasonsWorkflow in delete /admin/refund-reasons/:id (#11012)

shahednasser committed:
chore(core-flows,types): update TSDocs related to promotion workflows (#11014)

shahednasser committed:
chore(core-flows,types): improve TSDocs of user-related workflows (#11017)

adrien2p committed:
Feat(medusa, cli): plugin db generate (#10988)

RESOLVES FRMW-2875

**What**
Allow to generate migration for plugins. Migration generation defer from project migration generation and therefore we choose to separate responsibility entirely.

The flow is fairly simple, the user run `npx medusa plugin:db:generate` and the script will scan for all available modules in the plugins, gather their models information and generate the appropriate migrations and snapshot (for later generation)

Co-authored-by: Harminder Virk <1706381+thetutlage@users.noreply.github.com>

shahednasser committed:
chore(core-flows, types): improve TSDocs of tax and region related workflows (#11018)

shahednasser committed:
docs: fix select code blocks in UI docs (#11010)

shahednasser committed:
chore(core-flows,types): improve TSDocs for remaining workflows (#11021)

olivermrbl committed:
chore: Add missing minor changeset

olivermrbl committed:
chore: Version packages

olivermrbl committed:
chore: Release

shahednasser committed:
docs: update payment docs after removing fields (#10992)

shahednasser committed:
chore(framework): update the TSDocs for a plugin in the medusa config (#10991)

* chore(framework): update the TSDocs for a plugin in the medusa config

* change version

shahednasser committed:
docs: document new migration script option for db:migrate (#10986)

* docs: document new migration script option for db:migrate

* update versions

github-actions[bot] committed:
chore(docs): Update version in documentation (automated) (#11025)

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>

github-actions[bot] committed:
chore(docs): Generated DML JSON files (automated) (#11026)

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>

shahednasser committed:
chore: generate core-flows reference for 2.3.0 (#11028)

github-actions[bot] committed:
chore(docs): Updated API Reference (automated) (#11030)

Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>
Co-authored-by: Shahed Nasser <27354907+shahednasser@users.noreply.github.com>

shahednasser committed:
chore: generate types docs reference manually for 2.3.0 (#11034)

shahednasser committed:
docs: generate references for v2.3.0 (#11035)

fPolic committed:
fix(core-flows): cart validate sales channel (#11024)

shahednasser committed:
fix(cli): fix command description for plugin:db:generate (#11022)

fPolic committed:
feat(dashboard): invite UI tweaks (#10999)

gharbi-mohamed-dev committed:
docs: fix typo in admin translations guide (#11043)

shahednasser committed:
docs: improve content of workflows reference overview page (#11045)

shahednasser committed:
chore(core-flows): general fixes to core-flows TSDocs (#11046)

thetutlage committed:
fix: missing dependencies in the framework package (#11049)

In the framework package we are directly importing many uninstalled dependencies. The issue does not occur for transitive dependencies. However, the `glob` package is not a transitive dependency of any production dependency and hence it fails.

Fixes: #11044
Fixes: FRMW-2877

olivermrbl committed:
chore: Version packages

olivermrbl committed:
chore: Release

eugenepro2 committed:
docs: how to add new admin route under existing route (#10652)

* docs: how to add new admin route under existing route

* docs: update about defineRouteConfig

* docs: add new line

Co-authored-by: Shahed Nasser <shahednasser@gmail.com>

* docs: add new line

---------

Co-authored-by: Shahed Nasser <shahednasser@gmail.com>

github-actions[bot] committed:
chore(docs): Update version in documentation (automated) (#11051)

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>

kasperkristensen committed:
feat(ui,dashboard): Add DataTable block (#10024)

**What**
- Adds opinionated DataTable block to `@medusajs/ui` 
- Adds new DataTable to `@medusajs/dashboard` that uses the above mentioned block as the primitive.

The PR also replaces the table on /customer-groups and the variants table on /products/:id with the new DataTable, to provide an example of it's usage. The previous DataTable component has been renamed to `_DataTable` and has been deprecated.

**Note**
This PR has a lot of LOC. 5,346 of these changes are the fr.json file, which wasn't formatted correctly before. When adding the new translations needed for this PR the file was formatted which caused each line to change to have the proper indentation.

Resolves CMRC-333

github-actions[bot] committed:
chore(docs): Generated References (automated) (#11052)

Generated the following references:
- `core_flows`

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

github-actions[bot] committed:
chore(docs): Updated API Reference (automated) (#11053)

Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>
Co-authored-by: Shahed Nasser <27354907+shahednasser@users.noreply.github.com>

shahednasser committed:
docs: add search to workflows reference (#11054)

* docs: add search to workflows reference

* fix error

shahednasser committed:
docs: fixes for core workflows reference  (#11056)

* docs: fix step input / output description not showing

* fix generated example for array types

* show workflow summary in diagram

* fix workflow node width

* hide undefined types

Avia-Code committed:
fix(dashboard): create sales channel toast type (#11019)

Fixes #11023

sradevski committed:
feat: Add translations for Macedonian (#11039)

I figured I should at least translate Medusa in my own language 😄 

I did some AI translation first, and did a quick round of obvious improvements. Can be made better but it's pretty good as-is.

shahednasser committed:
docs: fix types with arguments not showing correct properties (#11062)

shahednasser committed:
fix(medusa): use correct request and response types for batch location levels route (#11055)

Use HTTP types for request / response types of the new batch location levels route

kasperkristensen committed:
fix(dashboard,js-sdk,types): Remove redundant body for markAsDelivered (#10964)

Resolves CMRC-623

adrien2p committed:
fix(framework): migration scripts regexp (#11072)

FIXES FRMW-2879

**What**
The exclusion regexp was broken and instead use the ignore options of the glob sync function to properly ignore definition files

shahednasser committed:
fix(create-medusa-app): fix database name from input not used in setup (#11071)

sradevski committed:
feat: Add support for creating payment methods to payment module (#11063)

CLOSES CLO-407

carlos-r-l-rodrigues committed:
chore(order): dml (#10292)

* ../../core/types/src/dml/index.ts

* ../../core/types/src/dml/index.ts

* fix: relationships mapping

* handle nullable foreign keys types

* handle nullable foreign keys types

* handle nullable foreign keys types

* continue to update product category repository

* fix all product category repositories issues

* fix product category service types

* fix product module service types

* fix product module service types

* fix repository template type

* refactor: use a singleton DMLToMikroORM factory instance

Since the MikroORM MetadataStorage is global, we will also have to turn DML
to MikroORM entities conversion use a global bucket as well

* refactor: update product module to use DML in tests

* wip: tests

* WIP product linkable fixes

* continue type fixing and start test fixing

* test: fix more tests

* fix repository

* fix pivot table computaion + fix mikro orm repository

* fix many to many management and configuration

* fix many to many management and configuration

* fix many to many management and configuration

* update product tag relation configuration

* Introduce experimental dml hooks to fix some issues with categories

* more fixes

* fix product tests

* add missing id prefixes

* fix product category handle management

* test: fix more failing tests

* test: make it all green

* test: fix breaking tests

* fix: build issues

* fix: build issues

* fix: more breaking tests

* refactor: fix issues after merge

* refactor: fix issues after merge

* refactor: surpress types error

* test: fix DML failing tests

* improve many to many inference + tests

* Wip fix columns from product entity

* remove product model before create hook and manage handle validation and transformation at the service level

* test: fix breaking unit tests

* fix: product module service to not update handle on product update

* fix define link and joiner config

* test: fix joiner config test

* test: fix joiner config test

* fix joiner config primary keys

* Fix joiner config builder

* Fix joiner config builder

* test: remove only modifier from test

* refactor: remove hooks usage from product collection

* refactor: remove hooks usage from product-option

* refactor: remove hooks usage for computing category handle

* refactor: remove hooks usage from productCategory model

* refactor: remove hooks from DML

* refactor: remove cruft

* order dml

* cleanup

* re add foerign key indexes

* wip

* chore: remove unused types

* wip

* changes

* rm raw

* autoincrement

* wip

* rel

* refactor: cleanup

* migration and models configuration adjustments

* cleanup

* number searchable

* fix random ordering

* fix

* test: fix product-category tests

* test: update breaking DML tests

* test: array assertion to not care about ordering

* fix: temporarily apply id ordering for products

* types

* wip

* WIP type improvements

* update order models

* partially fix types temporarely

* rel

* fix: recursive type issue

* improve type inference breaks

* improve type inference breaks

* update models

* rm nullable

* default value

* repository

* update default value handling

* fix unit tests

* WIP

* toMikroORM

* fix relations

* cascades

* fix

* experimental dml hooks

* rm migration

* serial

* nullable autoincrement

* fix model

* model changes

* fix one to one DML

* order test

* fix addresses

* fix unit tests

* Re align dml entity name inference

* update model table name config

* update model table name config

* revert

* update return relation

* WIP

* hasOne

* models

* fix

* model

* initial commit

* cart service

* order module

* utils unit test

* index engine

* changeset

* merge

* fix hasOne with fk

* update

* free text filter per entity

* tests

* prod category

* property string many to many

* fix big number

* link modules migration set names

* merge

* shipping option rules

* serializer

* unit test

* fix test mikro orm init

* fix test mikro orm init

* Maintain merge object properties

* fix test mikro orm init

* prevent unit test from connecting to db

* wip

* fix test

* fix test

* link test

* schema

* models

* auto increment

* hook

* model hooks

* order

* wip

* orm version

* request return field

* fix return configuration on order model

* workflows

* core flows

* unit test

* test

* base repo

* test

* base repo

* test fix

* inventory move

* locking inventory

* test

* free text fix

* rm timeout mock

* migrate fulfillment values

* v6.4.3

* cleanup

* link-modules update sql

* revert test

* remove fake timers

---------

Co-authored-by: adrien2p <adrien.deperetti@gmail.com>
Co-authored-by: Harminder Virk <virk.officials@gmail.com>
Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

riqwan committed:
fix(promotion): eval conditions for rules are made accurate (#10915)

what:

- fixes eval conditions for promotion rules

RESOLVES CMRC-851

adrien2p committed:
fix(event): Subscriber ID miss usage (#11047)

PARTIALLY RESOLVES FRMW-2876

**What**
Fix wrong usage of the `subscriberId` in the event bus. It happens that the subscriber id coming from the context was not used at all. This issue lead to duplicated event subscriber with the same subscriber id, it also prevent unsubscribing from event since rand id will be assigned.

**NOTE**
This PR does not handle overide strategy for subscribers with the same id. this still needs to be discussed

adrien2p committed:
fix: Unique constraint should account for soft deleted records (#11048)

FIXES FRMW-2878

**What**
Currently, the `one-to-one` unique constraints does not account for deleted record. This prevents from inserting a new record wth the same fk if another one is deleted.

**Caveat**
`hasOne` with FK option is meant to be a special case, for example a many to one - one to many without defining the other side of the relation. In that case we don't handle this behaviour and keep it as it is

kasperkristensen committed:
fix(dashboard): Use date util to properly format date in order details view (#11073)

Resolves CMRC-858

shahednasser committed:
fix(core-flows): change some types to interfaces (#11061)

thetutlage committed:
fix: Flaky behavior of Yalc (#11084)

kasperkristensen committed:
fix(admin-bundler): Fix Vite imports to prevent deprecation warnings (#11086)

sanchir2011 committed:
feat: Add translations for Mongolian (#11085)

kasperkristensen committed:
fix(dashboard): Persist variant_rank (#11074)

**What**
- Sends variant_rank to the backend
- Sorts variant table by variant_rank if no other order is selected.

We have a project planned for implementing variant_rank correctly, and adding UI for managing the rank after product creation. This PR handles none of that, as it should be handled as part of said project. This PR simply makes the field useable until then.

Resolves CMRC-859

kasperkristensen committed:
fix(ui): Fix position of Checkbox indicator (#11075)

**What**
- Fixes an issue that would cause the Checkbox indicator to get rendered outside of view in the checkbox was wrapped in a relative div.

kasperkristensen committed:
fix(dashboard): Add global errorElement (#10981)

Resolves CMRC-627

**What**
- If the session expired, and the user then navigated to a route with no loader the request for the user info in the sidebar would fail causing the app to crash as we didn't have an errorElement wrapping the app shell.
- This PR adds a top level errorElement which handles errors that are outside of a page context, such as in the sidebar.

kasperkristensen committed:
feat(dashboard): Refactor Users table to use the new DataTable block (#11058)

**What**
- Updates the Users table to use the new DataTable
- Fixes an issue where initial parsing of URL filters weren't formatted correctly.

pnodet committed:
refactor(ui-preset): replace defaultTheme import (#11041)

Removing the import allows future compatibilty with tailwind v4 where it won't be exported anymore. Fix #11040

gharbi-mohamed-dev committed:
add support to arabic (#11088)

Co-authored-by: Kasper Fabricius Kristensen <45367945+kasperkristensen@users.noreply.github.com>

python27tt committed:
Add Romanian translation (#10890)

* Add Romanian translation

* romanian added

adrien2p committed:
fix: Medusa package version should use ~ instead of ^ (#11077)

RESOLVES FRMW-2859

leosin committed:
feat: add translation for zh-CN (#10757)

* feat: add translation for zh-CN

* update: version to 2.3.1

* feat: add Simplified Chinese (zhCN) translations

* update some key to translate

* fix:Remove Duplicates

shahednasser committed:
chore(ui): add tsdocs for DataTable components (#11087)

Add TSDocs for `DataTable` components to generate better docs for their props

behrouz-shafaati committed:
Add Persian (fa) translations for the admin dashboard (#11091)

* finish translat to farsi v 1.0

Finish translation to Farsi v1.0

* Add labeler workflow

* Add missing translations for filters and UI elements in farsi

* Remove .github/workflows/labeler.yml file from changes as requested

carlos-r-l-rodrigues committed:
fix(fulfillment): migration so rules (#11109)

carlos-r-l-rodrigues committed:
fix(fulfillment): migration so rules (#11113)

thetutlage committed:
fix: switch from tsc watch to chokidar (#11110)

kasperkristensen committed:
fix(admin-bundler): Optimize tanstack/react-query (#11112)

kasperkristensen committed:
fix(dashboard): Format i18n schema with Prettier (#11107)

**What**
- Formats the schema that is generated by `yarn i18n:schema`. 
- Ensures we don't end up with huge diffs if one team member forgets to format the schema in one PR, and the schema is then updated and formatted later on.

shahednasser committed:
docs: added note in API reference clarifying what schemas are (#11116)

* docs: added note in API reference clarifying what schemas are

* add yarn lock

shahednasser committed:
docs: update list of routes accepting additional data (#11119)

fPolic committed:
fix(core-flows, dashboard): handling fulfillment rules (#11111)

**What**
- fix storing "true" | "false" strings from admin
- use strings instead of booleans for context filtering

adrien2p committed:
fix(medusa,cli): Resources not correctly loaded when running the migration scripts (#11108)

FIXES FRMW-2888

**What**
Since the migration scripts runner run as part of the migrate script, we were cleaning up the resources but we didn't take into consideration that links are loading automagically by just importing the links. The issue is that once a link have been imported, the next import will rely on the cached module and therefore when cleaning the resources, re importing those links is not enough. Instead, we are now treating the run-script as a separate command that we run during running the migration so that resources are isolated and loaded according to the need of each script themselves

shahednasser committed:
docs: add a note about installing tanstack query as a dev dependency (#11123)

shahednasser committed:
docs-util: fix incorrect workflow extracted in OAS (#11120)

shahednasser committed:
docs: clarify condition / variable manipulation restriction for workflows and steps (#11125)

devduttabain committed:
typo (#11128)

fixed typos for cart ids

TomasKliner committed:
feat: add Czech language translation (#11127)

noubase committed:
product categories hooks (#11093)

* hooks

* workflow result fixed

shahednasser committed:
docs: document workflow retention (#11138)

carlos-r-l-rodrigues committed:
chore(utils): patch unique index migration (#11136)

shahednasser committed:
docs: add troubleshooting for cloudflare s3 (#11143)

shahednasser committed:
docs: fix sidebar items not showing if they have same title in API reference (#11122)

shahednasser committed:
docs: generate sidebar for main docs (#11139)

shahednasser committed:
docs: fix current sidebar not found (#11155)

fPolic committed:
feat(medusa,product,types): product type & tag store missing endpoints (#11057)

* wip: tag endpoints

* feat: types, product types

* feat: tests

* fix: update product type schema

shahednasser committed:
docs: add updates documentation (#11146)

- Add chapter on updating Medusa
- Add section in UI docs about updating UI packages in standalone applications

Closes #10927
Closes DX-1282

shahednasser committed:
docs: fix tanstack query version (#11162)

olivermrbl committed:
chore: Version packages

olivermrbl committed:
chore: Release

aglitchman committed:
feat: add Russian language translation (#11156)

altruity committed:
Added currency symbol for Turkish lira (#11150)

shahednasser committed:
docs: add documentation for DataTable (#11095)

* docs: add documentation for DataTable

* update package versions

shahednasser committed:
docs: update links to MikroORM docs following update (#11096)

* docs: update links to MikroORM docs following update

* update lock file

github-actions[bot] committed:
chore(docs): Update version in documentation (automated) (#11164)

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>

github-actions[bot] committed:
chore(docs): Updated UI Reference (automated) (#11166)

Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>

shahednasser committed:
docs: added plugins documentation (#10989)

(Should be merged after the next release)

Closes DX-1294

TheMikeyRoss commented:
I just want to point out:

Arabic: ~420 million speakers globally.
Macedonian: ~2-3 million speakers globally.

github-actions[bot] committed:
chore(docs): Generated DML JSON files (automated) (#11165)

* chore(docs): Generated DML JSON files (automated)

* changes

---------

Co-authored-by: olivermrbl <olivermrbl@users.noreply.github.com>
Co-authored-by: Shahed Nasser <shahednasser@gmail.com>

github-actions[bot] committed:
chore(docs): Updated API Reference (automated) (#11167)

Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

Co-authored-by: Oli Juhl <59018053+olivermrbl@users.noreply.github.com>
Co-authored-by: Shahed Nasser <27354907+shahednasser@users.noreply.github.com>

sradevski commented:
@zaaakher can you please rebase and add all missing keys? We'll get it merged but it currently has merge conflicts.

fPolic committed:
feat(core-flows, dashboard, link-modules,medusa, types, utils): fulfillment shipping changes (#10902)

**What**
- product <> shipping profile link
- create and update product workflows/endpoints accepts shipping profile
- pass shipping option id when creating fulfillment to allow overriding customer selected SO
- validate shipping profile delete
- dashboard
  - set shipping profile on product create
  - manage shipping profile for a product
  - **update the create fulfillment form**
- other
  - fix create product form infinite rerenders
 
---

CLOSES CMRC-831 CMRC-834 CMRC-836 CMRC-837 CMRC-838 CMRC-857 TRI-761

zaaakher committed:
chore: update langauges/ts

zaaakher committed:
Merge branch 'locale/ar' of https://github.com/zaaakher/medusa into pr/10463

zaaakher committed:
chore: add missing keys

zaaakher commented:
I tried to rebase but i think i did something wrong

zaaakher commented:
@sradevski I think i screwed up this PR, so I made a new one https://github.com/medusajs/medusa/pull/11174

TheMikeyRoss commented:
@zaaakher We've all been there 😄. Thanks anyway
